<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Steven">

<title>Mini Lecture: An Explorative Analysis Based on A Research Paper</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="mini_lecture_files/libs/clipboard/clipboard.min.js"></script>
<script src="mini_lecture_files/libs/quarto-html/quarto.js"></script>
<script src="mini_lecture_files/libs/quarto-html/popper.min.js"></script>
<script src="mini_lecture_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="mini_lecture_files/libs/quarto-html/anchor.min.js"></script>
<link href="mini_lecture_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="mini_lecture_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="mini_lecture_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="mini_lecture_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="mini_lecture_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#load-ctcf-protein-sequence" id="toc-load-ctcf-protein-sequence" class="nav-link" data-scroll-target="#load-ctcf-protein-sequence">Load CTCF protein sequence</a></li>
  <li><a href="#calculate-the-arginine-and-lysine-frequency" id="toc-calculate-the-arginine-and-lysine-frequency" class="nav-link" data-scroll-target="#calculate-the-arginine-and-lysine-frequency">Calculate the arginine and lysine frequency</a>
  <ul class="collapse">
  <li><a href="#creat-a-function-to-calculate-the-frequency-of-arginine-and-lysine" id="toc-creat-a-function-to-calculate-the-frequency-of-arginine-and-lysine" class="nav-link" data-scroll-target="#creat-a-function-to-calculate-the-frequency-of-arginine-and-lysine">Creat a function to calculate the frequency of arginine and lysine</a></li>
  <li><a href="#a-quick-test" id="toc-a-quick-test" class="nav-link" data-scroll-target="#a-quick-test">A quick test</a></li>
  </ul></li>
  <li><a href="#calculate-the-arginine-and-lysine-frequency-for-the-ctcf-sequence" id="toc-calculate-the-arginine-and-lysine-frequency-for-the-ctcf-sequence" class="nav-link" data-scroll-target="#calculate-the-arginine-and-lysine-frequency-for-the-ctcf-sequence">Calculate the arginine and lysine frequency for the CTCF sequence</a></li>
  <li><a href="#retrieve-the-plddt-disordered-score-from-the-alphafold-predicted-pdb-file-of-ctcf" id="toc-retrieve-the-plddt-disordered-score-from-the-alphafold-predicted-pdb-file-of-ctcf" class="nav-link" data-scroll-target="#retrieve-the-plddt-disordered-score-from-the-alphafold-predicted-pdb-file-of-ctcf">Retrieve the pLDDT (disordered) score from the Alphafold predicted PDB file of CTCF</a>
  <ul class="collapse">
  <li><a href="#load-the-alphafold-predicted-pdb-file-of-ctcf-af-p49711-f1-model_v4.pdb-by-library-bio3d." id="toc-load-the-alphafold-predicted-pdb-file-of-ctcf-af-p49711-f1-model_v4.pdb-by-library-bio3d." class="nav-link" data-scroll-target="#load-the-alphafold-predicted-pdb-file-of-ctcf-af-p49711-f1-model_v4.pdb-by-library-bio3d.">Load the Alphafold predicted PDB file of CTCF “AF-P49711-F1-model_v4.pdb” by library <code>bio3d</code>.</a></li>
  <li><a href="#retrieve-the-b-factor-of-each-c-elety-from-the-pdb-file-use-library-dplyr" id="toc-retrieve-the-b-factor-of-each-c-elety-from-the-pdb-file-use-library-dplyr" class="nav-link" data-scroll-target="#retrieve-the-b-factor-of-each-c-elety-from-the-pdb-file-use-library-dplyr">Retrieve the B-factor of each “C” elety from the PDB file use library <code>dplyr</code></a></li>
  <li><a href="#cbind-the-b-factor-to-the-ctcf_kr_freq-dataframe" id="toc-cbind-the-b-factor-to-the-ctcf_kr_freq-dataframe" class="nav-link" data-scroll-target="#cbind-the-b-factor-to-the-ctcf_kr_freq-dataframe">cbind the B-factor to the CTCF_KR_freq dataframe</a></li>
  <li><a href="#make-a-plot" id="toc-make-a-plot" class="nav-link" data-scroll-target="#make-a-plot">Make a plot</a></li>
  </ul></li>
  <li><a href="#analyse-the-ctcf-rna-immunoprecipitation-sequencing-rip-seq-data-from-encode" id="toc-analyse-the-ctcf-rna-immunoprecipitation-sequencing-rip-seq-data-from-encode" class="nav-link" data-scroll-target="#analyse-the-ctcf-rna-immunoprecipitation-sequencing-rip-seq-data-from-encode">Analyse the CTCF RNA immunoprecipitation sequencing (RIP-seq) data from ENCODE</a>
  <ul class="collapse">
  <li><a href="#load-the-gene-quantification-tsv-file-encff552mwl.tsv-by-library-readr" id="toc-load-the-gene-quantification-tsv-file-encff552mwl.tsv-by-library-readr" class="nav-link" data-scroll-target="#load-the-gene-quantification-tsv-file-encff552mwl.tsv-by-library-readr">Load the gene quantification TSV file “ENCFF552MWL.tsv” by library <code>readr</code></a></li>
  <li><a href="#have-a-look-at-this-dataset" id="toc-have-a-look-at-this-dataset" class="nav-link" data-scroll-target="#have-a-look-at-this-dataset">Have a look at this dataset</a></li>
  <li><a href="#filter-the-dataset-to-remove-rows-with-na-inf-or-less-than-0-log2fold_change-values-and-q_value-values-of-1." id="toc-filter-the-dataset-to-remove-rows-with-na-inf-or-less-than-0-log2fold_change-values-and-q_value-values-of-1." class="nav-link" data-scroll-target="#filter-the-dataset-to-remove-rows-with-na-inf-or-less-than-0-log2fold_change-values-and-q_value-values-of-1.">Filter the dataset to remove rows with NA, Inf, or less than 0 <code>log2(fold_change)</code> values and <code>q_value</code> values of 1.</a></li>
  <li><a href="#lets-plot-the-log2-fold-change-vs-the---log10-q-value" id="toc-lets-plot-the-log2-fold-change-vs-the---log10-q-value" class="nav-link" data-scroll-target="#lets-plot-the-log2-fold-change-vs-the---log10-q-value">Let’s plot the log2 fold change vs the - log10 q-value</a></li>
  <li><a href="#print-genes-name-that-have-a-log2fold_change-2-and-are-significant" id="toc-print-genes-name-that-have-a-log2fold_change-2-and-are-significant" class="nav-link" data-scroll-target="#print-genes-name-that-have-a-log2fold_change-2-and-are-significant">Print genes name that have a <code>log2(fold_change)</code> &gt; 2 and are significant</a></li>
  </ul></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info">Session Info</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Mini Lecture: An Explorative Analysis Based on A Research Paper</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Steven </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>Let’s analyse a recent <a href="https://www.sciencedirect.com/science/article/pii/S1097276523004343?via%3Dihub">paper</a> studying transcription factors interacting with RNA</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="fig1.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure 1: Graphical abstract</figcaption>
</figure>
</div>
<p>The paper emphasize the role of arginine- and lysine-rich motif (ARM) in the interaction between transcription factors and RNA. The authors found that the vast majority of RNA-binding transcription factors contain an ARM, and that ARM is a key determinant of RNA-binding specificity and affinity.</p>
<p>I am interested in the arginine and lysine frequency in the CTCF protein sequence. CTCF is a transcription factor that binds to DNA and RNA. I want to see if CTCF has ARMs and where they are located.</p>
</section>
<section id="load-ctcf-protein-sequence" class="level1">
<h1>Load CTCF protein sequence</h1>
<p>Let’s first view the protein information of <a href="https://www.uniprot.org/uniprotkb/P49711/entry">CTCF</a> on the <a href="https://www.uniprot.org/">UniProt</a> database and get the protein sequence.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>CTCF_seq <span class="ot">&lt;-</span> <span class="st">"MEGDAVEAIVEESETFIKGKERKTYQRRREGGQEEDACHLPQNQTDGGEVVQDVNSSVQMVMMEQLDPTLLQMKTEVMEGTVAPEAEAAVDDTQIITLQVVNMEEQPINIGELQLVQVPVPVTVPVATTSVEELQGAYENEVSKEGLAESEPMICHTLPLPEGFQVVKVGANGEVETLEQGELPPQEDPSWQKDPDYQPPAKKTKKTKKSKLRYTEEGKDVDVSVYDFEEEQQEGLLSEVNAEKVVGNMKPPKPTKIKKKGVKKTFQCELCSYTCPRRSNLDRHMKSHTDERPHKCHLCGRAFRTVTLLRNHLNTHTGTRPHKCPDCDMAFVTSGELVRHRRYKHTHEKPFKCSMCDYASVEVSKLKRHIRSHTGERPFQCSLCSYASRDTYKLKRHMRTHSGEKPYECYICHARFTQSGTMKMHILQKHTENVAKFHCPHCDTVIARKSDLGVHLRKQHSYIEQGKKCRYCDAVFHERYALIQHQKSHKNEKRFKCDQCDYACRQERHMIMHKRTHTGEKPYACSHCDKTFRQKQLLDMHFKRYHDPNFVPAAFVCSKCGKTFTRRNTMARHADNCAGPDGVEGENGGETKKSKRGRKRKMRSKKEDSSDSENAEPDLDDNEDEEEPAVEIEPEPEPQPVTPAPPPAKKRRGRPPGRTNQPKQNQPTAIIQVEDQNTGAIENIIVEVKKEPDAEPAEGEEEEAQPAATDAPNGDLTPEMILSMMDR"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calculate-the-arginine-and-lysine-frequency" class="level1">
<h1>Calculate the arginine and lysine frequency</h1>
<section id="creat-a-function-to-calculate-the-frequency-of-arginine-and-lysine" class="level2">
<h2 class="anchored" data-anchor-id="creat-a-function-to-calculate-the-frequency-of-arginine-and-lysine">Creat a function to calculate the frequency of arginine and lysine</h2>
<p>I want to calculate the frequency of arginine and lysine in the CTCF sequence by creating a function in R. The function will take in a sequence and return the frequency of arginine and lysine.</p>
<ol type="1">
<li>Convert the sequence to uppercase</li>
<li>Set the chunk size to 5 AA</li>
<li>Scan the whole sequence by moving the chunk by 1 AA and calculate the frequency of arginine and lysine for each chunk.</li>
<li>Store the frequency in a dataframe with columns: site, R_freq, K_freq, RK_freq.</li>
</ol>
<p>Prompt it into GenAI ~</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>count_RK_freq <span class="ot">&lt;-</span> <span class="cf">function</span>(sequence) {</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert the sequence to uppercase</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  sequence <span class="ot">&lt;-</span> <span class="fu">toupper</span>(sequence)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set the chunk size</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  chunk_size <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Initialize a dataframe to store the results</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">site =</span> <span class="fu">integer</span>(), <span class="at">R_freq =</span> <span class="fu">numeric</span>(), <span class="at">K_freq =</span> <span class="fu">numeric</span>(), <span class="at">RK_freq =</span> <span class="fu">numeric</span>())</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Scan the sequence</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">nchar</span>(sequence) <span class="sc">-</span> chunk_size <span class="sc">+</span> <span class="dv">1</span>)) {</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    chunk <span class="ot">&lt;-</span> <span class="fu">substr</span>(sequence, i, i <span class="sc">+</span> chunk_size <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate frequencies</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>    r_freq <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">strsplit</span>(chunk, <span class="st">""</span>)[[<span class="dv">1</span>]] <span class="sc">==</span> <span class="st">"R"</span>) <span class="sc">/</span> chunk_size</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>    k_freq <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">strsplit</span>(chunk, <span class="st">""</span>)[[<span class="dv">1</span>]] <span class="sc">==</span> <span class="st">"K"</span>) <span class="sc">/</span> chunk_size</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>    rk_freq <span class="ot">&lt;-</span> r_freq <span class="sc">+</span> k_freq</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add to dataframe</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results, <span class="fu">data.frame</span>(<span class="at">site =</span> i, <span class="at">R_freq =</span> r_freq, <span class="at">K_freq =</span> k_freq, <span class="at">RK_freq =</span> rk_freq))</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(results)</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="a-quick-test" class="level2">
<h2 class="anchored" data-anchor-id="a-quick-test">A quick test</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>test_seq <span class="ot">&lt;-</span> <span class="st">"mmmmmmmmmRRrKKKKnnnnnnnnRnnnnKnnn"</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>test_seq <span class="ot">&lt;-</span> <span class="fu">count_RK_freq</span>(test_seq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(test_seq) <span class="sc">+</span> <span class="fu">aes</span>(<span class="at">x =</span> site, <span class="at">y =</span> RK_freq) <span class="sc">+</span> <span class="fu">geom_line</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="mini_lecture_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="calculate-the-arginine-and-lysine-frequency-for-the-ctcf-sequence" class="level1">
<h1>Calculate the arginine and lysine frequency for the CTCF sequence</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>CTCF_KR_freq <span class="ot">&lt;-</span> <span class="fu">count_RK_freq</span>(CTCF_seq)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot KR_freq vs site by geom_area</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(CTCF_KR_freq) <span class="sc">+</span> </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> site, <span class="at">y =</span> RK_freq) <span class="sc">+</span> </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_area</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="mini_lecture_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot K_freq, R_freq, and KR_freq vs site by geom_area filled with different colors</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(CTCF_KR_freq) <span class="sc">+</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_area</span>(<span class="fu">aes</span>(<span class="at">x =</span> site, <span class="at">y =</span> RK_freq, <span class="at">fill =</span> <span class="st">"RK"</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_area</span>(<span class="fu">aes</span>(<span class="at">x =</span> site, <span class="at">y =</span> K_freq, <span class="at">fill =</span> <span class="st">"K"</span>), <span class="at">alpha =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_area</span>(<span class="fu">aes</span>(<span class="at">x =</span> site, <span class="at">y =</span> R_freq, <span class="at">fill =</span> <span class="st">"R"</span>), <span class="at">alpha =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add a legend</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"AA"</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"RK"</span> <span class="ot">=</span> <span class="st">"black"</span>, <span class="st">"K"</span> <span class="ot">=</span> <span class="st">"lightpink"</span>, <span class="st">"R"</span> <span class="ot">=</span> <span class="st">"lightblue"</span>)) <span class="sc">+</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add a title and labels</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Arginine and Lysine Frequency in CTCF"</span>, <span class="at">x =</span> <span class="st">"Site"</span>, <span class="at">y =</span> <span class="st">"Frequency"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="mini_lecture_files/figure-html/unnamed-chunk-5-2.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="retrieve-the-plddt-disordered-score-from-the-alphafold-predicted-pdb-file-of-ctcf" class="level1">
<h1>Retrieve the pLDDT (disordered) score from the Alphafold predicted PDB file of CTCF</h1>
<section id="load-the-alphafold-predicted-pdb-file-of-ctcf-af-p49711-f1-model_v4.pdb-by-library-bio3d." class="level2">
<h2 class="anchored" data-anchor-id="load-the-alphafold-predicted-pdb-file-of-ctcf-af-p49711-f1-model_v4.pdb-by-library-bio3d.">Load the Alphafold predicted PDB file of CTCF “AF-P49711-F1-model_v4.pdb” by library <code>bio3d</code>.</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bio3d)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>CTCF_pdb <span class="ot">&lt;-</span> <span class="fu">read.pdb</span>(<span class="st">"AF-P49711-F1-model_v4.pdb"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="retrieve-the-b-factor-of-each-c-elety-from-the-pdb-file-use-library-dplyr" class="level2">
<h2 class="anchored" data-anchor-id="retrieve-the-b-factor-of-each-c-elety-from-the-pdb-file-use-library-dplyr">Retrieve the B-factor of each “C” elety from the PDB file use library <code>dplyr</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>CTCF_b <span class="ot">&lt;-</span> CTCF_pdb<span class="sc">$</span>atom <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(elety <span class="sc">==</span> <span class="st">"CA"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(b)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cbind-the-b-factor-to-the-ctcf_kr_freq-dataframe" class="level2">
<h2 class="anchored" data-anchor-id="cbind-the-b-factor-to-the-ctcf_kr_freq-dataframe">cbind the B-factor to the CTCF_KR_freq dataframe</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># CTCF_KR_freq &lt;- cbind(CTCF_KR_freq, CTCF_b)</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co"># This code does not work because the number of rows of CTCF_KR_freq and CTCF_b are different.</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># We need to trimmed the first and last two elements of CTCF_b to reach the same length as CTCF_KR_freq</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>CTCF_b <span class="ot">&lt;-</span> CTCF_b[<span class="dv">3</span><span class="sc">:</span>(<span class="fu">nrow</span>(CTCF_b) <span class="sc">-</span> <span class="dv">2</span>), ]</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>CTCF_KR_freq <span class="ot">&lt;-</span> <span class="fu">cbind</span>(CTCF_KR_freq, CTCF_b)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="make-a-plot" class="level2">
<h2 class="anchored" data-anchor-id="make-a-plot">Make a plot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(CTCF_KR_freq) <span class="sc">+</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> site, <span class="at">y =</span> CTCF_b <span class="sc">/</span> <span class="dv">100</span>), <span class="at">color =</span> <span class="st">"gray"</span>) <span class="sc">+</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_area</span>(<span class="fu">aes</span>(<span class="at">x =</span> site, <span class="at">y =</span> RK_freq, <span class="at">fill =</span> <span class="st">"RK"</span>), <span class="at">alpha =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_area</span>(<span class="fu">aes</span>(<span class="at">x =</span> site, <span class="at">y =</span> K_freq, <span class="at">fill =</span> <span class="st">"K"</span>), <span class="at">alpha =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_area</span>(<span class="fu">aes</span>(<span class="at">x =</span> site, <span class="at">y =</span> R_freq, <span class="at">fill =</span> <span class="st">"R"</span>), <span class="at">alpha =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"AA"</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"RK"</span> <span class="ot">=</span> <span class="st">"darkblue"</span>, <span class="st">"K"</span> <span class="ot">=</span> <span class="st">"lightpink"</span>, <span class="st">"R"</span> <span class="ot">=</span> <span class="st">"lightblue"</span>)) <span class="sc">+</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Arginine and Lysine Frequency in CTCF"</span>, <span class="at">x =</span> <span class="st">"Site"</span>, <span class="at">y =</span> <span class="st">"Frequency"</span>) <span class="sc">+</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw a line representing the protein sequence at the bottom of the plot</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="fu">nchar</span>(CTCF_seq), <span class="at">y =</span> <span class="dv">0</span>, <span class="at">yend =</span> <span class="dv">0</span>), <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Annotate the zinc finger regions of CTCF at sites: 266-577 by rectangles</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">geom_rect</span>(<span class="fu">aes</span>(<span class="at">xmin =</span> <span class="dv">266</span>, <span class="at">xmax =</span> <span class="dv">577</span>, <span class="at">ymin =</span> <span class="sc">-</span><span class="fl">0.025</span>, <span class="at">ymax =</span> <span class="fl">0.025</span>), <span class="at">fill =</span> <span class="st">"orange"</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="mini_lecture_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make a plot of KR_freq vs site with the B-factor as the color using area</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(CTCF_KR_freq) <span class="sc">+</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> site, <span class="at">y =</span> RK_freq, <span class="at">color =</span> <span class="fu">log</span>(CTCF_b) <span class="sc">^</span> (<span class="sc">-</span><span class="dv">1</span>))) <span class="sc">+</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Arginine and Lysine Frequency in CTCF"</span>, <span class="at">x =</span> <span class="st">"Site"</span>, <span class="at">y =</span> <span class="st">"Frequency"</span>) <span class="sc">+</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="mini_lecture_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="analyse-the-ctcf-rna-immunoprecipitation-sequencing-rip-seq-data-from-encode" class="level1">
<h1>Analyse the CTCF RNA immunoprecipitation sequencing (RIP-seq) data from ENCODE</h1>
<p>Let’s have a search and see if we have any CTCF RNA binding profile data.</p>
<p><a href="https://www.encodeproject.org/">ENCODE</a> is a public database of genomic data. It contains many types of genomic data, including RNA-seq, ChIP-seq, RIP-seq, etc.</p>
<p>RIP-seq data of CTCF is reported on K562 and GM12878 cell line. We will use the <a href="https://www.encodeproject.org/experiments/ENCSR433RRS/">CTCF RIP-seq data</a> from K562 cell line.</p>
<section id="load-the-gene-quantification-tsv-file-encff552mwl.tsv-by-library-readr" class="level2">
<h2 class="anchored" data-anchor-id="load-the-gene-quantification-tsv-file-encff552mwl.tsv-by-library-readr">Load the gene quantification TSV file “ENCFF552MWL.tsv” by library <code>readr</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>CTCF_RIP <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"ENCFF552MWL.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 48453 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (8): test_id, gene_id, gene, locus, sample_1, sample_2, status, significant
dbl (6): value_1, value_2, log2(fold_change), test_stat, p_value, q_value

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
</section>
<section id="have-a-look-at-this-dataset" class="level2">
<h2 class="anchored" data-anchor-id="have-a-look-at-this-dataset">Have a look at this dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(CTCF_RIP)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 14
  test_id           gene_id gene  locus sample_1 sample_2 status value_1 value_2
  &lt;chr&gt;             &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt;
1 ENSG00000000003.… ENSG00… TSPA… chrX… ctcf     input    NOTEST  0.0671  0.163 
2 ENSG00000000005.5 ENSG00… TNMD  chrX… ctcf     input    NOTEST  0       0.0454
3 ENSG00000000419.8 ENSG00… DPM1  chr2… ctcf     input    OK     63.7    42.7   
4 ENSG00000000457.8 ENSG00… SCYL3 chr1… ctcf     input    OK      8.75    8.78  
5 ENSG00000000460.… ENSG00… C1or… chr1… ctcf     input    OK     27.9    23.6   
6 ENSG00000000938.8 ENSG00… FGR   chr1… ctcf     input    NOTEST  0.0905  0.0420
# ℹ 5 more variables: `log2(fold_change)` &lt;dbl&gt;, test_stat &lt;dbl&gt;,
#   p_value &lt;dbl&gt;, q_value &lt;dbl&gt;, significant &lt;chr&gt;</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(CTCF_RIP<span class="sc">$</span>significant)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   no   yes 
46719  1734 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(CTCF_RIP<span class="sc">$</span>p_value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00005 0.49055 1.00000 0.74877 1.00000 1.00000 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(CTCF_RIP<span class="sc">$</span>q_value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
0.001922 0.813944 1.000000 0.835909 1.000000 1.000000 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Count how many genes are significant</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(CTCF_RIP<span class="sc">$</span>significant <span class="sc">==</span> <span class="st">"yes"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1734</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Count how many genes have a q-value &lt; 0.05</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(CTCF_RIP<span class="sc">$</span>q_value <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1734</code></pre>
</div>
</div>
</section>
<section id="filter-the-dataset-to-remove-rows-with-na-inf-or-less-than-0-log2fold_change-values-and-q_value-values-of-1." class="level2">
<h2 class="anchored" data-anchor-id="filter-the-dataset-to-remove-rows-with-na-inf-or-less-than-0-log2fold_change-values-and-q_value-values-of-1.">Filter the dataset to remove rows with NA, Inf, or less than 0 <code>log2(fold_change)</code> values and <code>q_value</code> values of 1.</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Trimmed out rows with NA or Inf `log2(fold_change)` values.</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>CTCF_RIP_trimmed <span class="ot">&lt;-</span> CTCF_RIP <span class="sc">%&gt;%</span> </span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(<span class="st">`</span><span class="at">log2(fold_change)</span><span class="st">`</span>) <span class="sc">&amp;</span> </span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>           <span class="st">`</span><span class="at">log2(fold_change)</span><span class="st">`</span> <span class="sc">!=</span> <span class="cn">Inf</span> <span class="sc">&amp;</span> </span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>           <span class="st">`</span><span class="at">log2(fold_change)</span><span class="st">`</span> <span class="sc">!=</span> <span class="sc">-</span><span class="cn">Inf</span> <span class="sc">&amp;</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>           <span class="st">`</span><span class="at">log2(fold_change)</span><span class="st">`</span> <span class="sc">&gt;=</span> <span class="dv">0</span>)</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Trimmed out rows with `q_value` values of 1.</span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>CTCF_RIP_trimmed <span class="ot">&lt;-</span> CTCF_RIP_trimmed <span class="sc">%&gt;%</span> </span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(q_value <span class="sc">!=</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="lets-plot-the-log2-fold-change-vs-the---log10-q-value" class="level2">
<h2 class="anchored" data-anchor-id="lets-plot-the-log2-fold-change-vs-the---log10-q-value">Let’s plot the log2 fold change vs the - log10 q-value</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(CTCF_RIP_trimmed) <span class="sc">+</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">log2(fold_change)</span><span class="st">`</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(q_value), <span class="at">color =</span> significant))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="mini_lecture_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot only the significant genes</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>CTCF_RIP_trimmed_sig <span class="ot">&lt;-</span> CTCF_RIP_trimmed <span class="sc">%&gt;%</span> </span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(significant <span class="sc">==</span> <span class="st">"yes"</span>)</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(CTCF_RIP_trimmed_sig) <span class="sc">+</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">log2(fold_change)</span><span class="st">`</span>, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(q_value), <span class="at">color =</span> <span class="st">"red"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="mini_lecture_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="print-genes-name-that-have-a-log2fold_change-2-and-are-significant" class="level2">
<h2 class="anchored" data-anchor-id="print-genes-name-that-have-a-log2fold_change-2-and-are-significant">Print genes name that have a <code>log2(fold_change)</code> &gt; 2 and are significant</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>CTCF_RIP_trimmed_sig <span class="sc">%&gt;%</span> </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">log2(fold_change)</span><span class="st">`</span> <span class="sc">&gt;</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(gene, <span class="st">`</span><span class="at">log2(fold_change)</span><span class="st">`</span>, q_value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 11 × 3
   gene          `log2(fold_change)` q_value
   &lt;chr&gt;                       &lt;dbl&gt;   &lt;dbl&gt;
 1 RP11-782C8.5                 2.91 0.0442 
 2 AC063976.1                   2.13 0.0244 
 3 SUCLA2P2                     2.03 0.00192
 4 CTC-261N6.3                  2.40 0.0450 
 5 CTD-2218K11.2                2.22 0.0317 
 6 CTD-2024D23.1                2.06 0.00192
 7 RP11-574M7.1                 2.60 0.00192
 8 RP11-325E5.1                 2.14 0.0228 
 9 CTC-448F2.4                  2.01 0.00192
10 CTD-2528L19.6                2.22 0.00192
11 VN1R83P                      2.08 0.00955</code></pre>
</div>
</div>
</section>
</section>
<section id="session-info" class="level1">
<h1>Session Info</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.1.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] readr_2.1.4   dplyr_1.1.4   bio3d_2.4-4   ggplot2_3.4.4

loaded via a namespace (and not attached):
 [1] bit_4.0.5         gtable_0.3.4      jsonlite_1.8.8    crayon_1.5.2     
 [5] compiler_4.3.2    tidyselect_1.2.0  Rcpp_1.0.11       parallel_4.3.2   
 [9] scales_1.3.0      yaml_2.3.7        fastmap_1.1.1     R6_2.5.1         
[13] labeling_0.4.3    generics_0.1.3    knitr_1.45        htmlwidgets_1.6.3
[17] tibble_3.2.1      munsell_0.5.0     pillar_1.9.0      tzdb_0.4.0       
[21] rlang_1.1.2       utf8_1.2.4        xfun_0.41         bit64_4.0.5      
[25] cli_3.6.1         withr_2.5.2       magrittr_2.0.3    digest_0.6.33    
[29] grid_4.3.2        vroom_1.6.4       rstudioapi_0.15.0 hms_1.1.3        
[33] lifecycle_1.0.4   vctrs_0.6.5       evaluate_0.23     glue_1.6.2       
[37] farver_2.1.1      fansi_1.0.5       colorspace_2.1-0  rmarkdown_2.25   
[41] tools_4.3.2       pkgconfig_2.0.3   htmltools_0.5.7  </code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>